### PR TITLE
Add an electric sports car

### PR DESCRIPTION
**Pull request by John-Candlebury**
God forbid you hit anything while driving this.

### CODE REVIEW

File: data/json/items/vehicle_parts.json [hunk 1]
```diff
@@ -254,6 +254,22 @@
     },
     {
         "type":"GENERIC",
+        "id" : "motor_enhanced",
+        "name" : "Enhanced electric motor",
+        "description" : "A very powerful and yet lightweight electric motor. Useful for crafting.",
```
Comment (illi-kun) [inline data/json/items/vehicle_parts.json]: Please put another one whitespace before "Useful for crafting".

**Comment by Coolthulhu** [comment]
The engine looks good to me. A fair bit too light (1/9 the weight of V12 for 1/4 the power), but compensates for it with efficiency ~300 epower/power, compared to large/medium electric engines' ~400 epower/power.
Would probably make perfect loot-bike engine due to good power/weight ratio.

But... does it have FOUR of those?

**Comment by Zireael07** [comment]
Great!

**Comment by LordTy** [comment]
The four engines is interesting, Are you trying to simulate hub-engines?
Wouldn't those be installed inside of the wheel?

**Comment by John-Candlebury** [comment]
>The four engines is interesting, Are you trying to simulate hub-engines?
 Wouldn't those be installed inside of the wheel?

Well kind of, I got the idea from the electric version of the Mercedes Sls Amg, which hasn't got hub motors, but does have a motor powering each individual wheel. 

And well the motors are mounted on the same tile as the wheel, actually.

**Comment by John-Candlebury** [comment]
Made a second version, this one is more or less an atomic ferrari. Tell me if its too stupid, or if you have a better name for a Ferrari/Lamborghini expy.

Doesnt really spawn anywhere yet, but I have this mansion I've been working work. Plus its not like we didnt had cars that never spawn already (the bubble car for example).

**Comment by Coolthulhu** [comment]
The atomic name is rather silly, yes. How about something generic, like Atomic Sports Car? It already sounds futuristic enough.

Also, those 4 engines per car are no-go. 1 is totally fine, 2 if you're feeling really generous.

The engines are supposed to be very light (superalloy), so you could make rest of the car super light too, by using extra-light parts rather than regular frames and quarterpanels (and doors).

**Comment by John-Candlebury** [comment]
They've got 4 so that they can compare to the rather ridiculous top speed of the normal sports car (form 300mph up to 400mph) and this cars have 320mph for the electric and 310mph for the atomic.

If the issue is that the cars are very good for parts,  the engines are not really that awesome individually, they are only slightly more efficient that a single large electric engine  and are technically slower.

The normal version is all lightweight frames already, and it doesn't really affect performance much. Engine weight doesn't really matter much either, even though the engines weight a fourth of the large version, the cars would still be faster with 4 large motors, they would be considerably less efficient however.

I can also simply make a single simple electric engine in the front, but I like the smaller engines in the wheels, they make the car considerably more fragile.

**Comment by Coolthulhu** [comment]
Oops, looks like I misread the values (swapped the sides of division). Sports engine should not be more efficient than the large one. It can be more powerful, but it should be less efficient, not more. Also, at <300 epower/power, it technically has >100% efficiency ratio (373 is 100%).

If you want to make the car fragile, having the engines in the front is enough. Back of the car is mostly safe, so adding engines there doesn't change much.

Current version has a total engine power of ~750. If you gave the single engine power of 470 and put 2 of those in the car, it would have about the same power as the current version.

Having a 4 engine car could be cool (because that would be a car you'd build your car around instead of just stripping for parts), but the current system makes such a configuration very inefficient.

**Comment by John-Candlebury** [comment]
Okay done, guess that the one engine per wheel idea can be considered  if we ever implement drivetrains and the like.

Perhaps once we got more car variation in the game it can be okay to give them names?
Would like to have non generic cars around someday.

**Comment by Coolthulhu** [comment]
Once someone writes a system for car models with tiny variations like single different door part etc. then yeah.
For now every car model is nearly unique (the only variation is part sizes), so the names should be informative before they become flavorful.

**Pull request merged**